api/services/tools_manage_service.py

```diff
@@ -0,0 +1,520 @@
+from typing import List, Tuple
+
+from flask import current_app
+
+from core.tools.tool_manager import ToolManager
+from core.tools.entities.user_entities import UserToolProvider, UserTool
+from core.tools.entities.tool_entities import ApiProviderSchemaType, ApiProviderAuthType, ToolProviderCredentials, \
+    ToolCredentialsOption
+from core.tools.entities.common_entities import I18nObject
+from core.tools.entities.tool_bundle import ApiBasedToolBundle
+from core.tools.provider.tool_provider import ToolProviderController
+from core.tools.provider.api_tool_provider import ApiBasedToolProviderController
+from core.tools.utils.parser import ApiBasedToolSchemaParser
+from core.tools.utils.encoder import serialize_base_model_array, serialize_base_model_dict
+from core.tools.utils.configration import ToolConfiguration
+from core.tools.errors import ToolProviderCredentialValidationError, ToolProviderNotFoundError, ToolNotFoundError
+
+from extensions.ext_database import db
+from models.tools import BuiltinToolProvider, ApiToolProvider
+
+from httpx import get
+
+import json
+
+class ToolManageService:
```

Comment:
@Yeuoly When I upload a large API spec like [this from Jira Cloud](https://developer.atlassian.com/cloud/jira/platform/swagger-v3.v3.json), I get a bad request 400. Do you know why? Is it too large for the server to parse? I don't see any error logs.

Reply:
resolved.

Reply:
Yup, but it would not allow me to save it "the number of apis should be less than 10". Could we get a checkbox in the frontend to select up to 10 functions from larger api specs?